rfcs/20191127-pip-structure.md

```diff
@@ -0,0 +1,212 @@
+# Title of RFC
+
+| Status        | (Proposed / Accepted / Implemented / Obsolete)       |
+:-------------- |:---------------------------------------------------- |
+| **RFC #**     | [NNN](https://github.com/tensorflow/community/pull/NNN) (update when you have community PR #)|
+| **Author(s)** | Anna Revinskaya (annarev@google.com)                 |
+| **Sponsor**   | Alex Passos (apassos@tensorflow.org)                 |
+| **Updated**   | 2019-11-27                                           |
+
+## Objective
+
+We propose to simplify TensorFlow pip package structure to enable IDE features such as autocomplete, jump-to-definition and quick-documentation.
```

Comment:
Another IDE feature that is missing due to the current packaging is the ability to verify compliance with abstract base classes. I believe this should be taken care of by the proposal, but it's worth verifying as a useful feature. 

E.g. when implementing a custom keras metric, your IDE should tell you that you're missing `update_state`:
https://github.com/tensorflow/tensorflow/blob/ddca4b92b4eb4e187716b03f443362c904bdad7e/tensorflow/python/keras/metrics.py#L215

Reply:
Sorry for not addressing this comment sooner. I just tried it out today.
New version of PyCharms would give abstract class warning out of the box. The old version that I tested while writing this doc (2019.1.1) would not give the warning and the change proposed in this RFC would fix it.

I tried in VS Code where some autocomplete does not work (tf.compat.v1.keras.) but could not find abstractclass warning even if I define a sample repro example.